fix file format for WMO files (*.rec2 and *.svs files)

QiShi-NOAA opened the pull request:
<b>Note to developers: You must use this PR template!</b>

## Description of Changes
> Please include a summary of the changes and the related GitHub issue(s). Please also include relevant motivation and context.

1. Added mandatory key “n” which hold the number of observations used to compute the given score to *.rec2 files
2. In the surface score files (*.svs files), changed the key for parameter name from “parm” to “par”."

## Developer Questions and Checklist
* Is this a high priority PR? If so, why and is there a date it needs to be merged by? NO
* Do you have any planned upcoming annual leave/PTO? NO
* Are there any changes needed in the times when the jobs are supposed to run/kick-off? NO
  
- [x] The code changes follow [NCO's EE2 Standards](https://www.nco.ncep.noaa.gov/idsb/implementation_standards/ImplementationStandards.v11.0.0.pdf).
- [x] Developer's name is removed throughout the code and have used `${USER}` where necessary throughout the code.
- [x] References the feature branch for `HOMEevs` are removed from the code.
- [x] J-Job environment variables, COMIN and COMOUT directories, and output follow what has been [defined](https://docs.google.com/document/d/1JWg_4q80aYmmAoD21GFjp9R9y5-3w7WGM3-0HJk0Pjs/edit#heading=h.7ysbr191vzu4) for EVS.
- [x] Jobs over 15 minutes in runtime have restart capability.
- [x] If applicable, changes in the `dev/drivers/scripts` or `dev/modulefiles` have been made in the corresponding `ecf/scripts` and `ecf/defs/evs-nco.def`? 
- [x] Jobs contain the appropriate file checking and don't run METplus for any missing data.
- [x] Code is using METplus wrappers structure and not calling MET executables directly.
- [x] Log is free of any ERRORs or WARNINGs.

## Testing Instructions

> Please include testing instructions for the PR assignee. Include all relevant input datasets needed to run the tests.

git clone https://github.com/QiShi-NOAA/EVS.git
cd EVS/
git checkout feature/wmo
ln -sf /lfs/h2/emc/vpppg/noscrub/emc.vpppg/verification/EVS_fix fix
cd sorc/
./build.sh

In  jevs_global_det_gfs_atmos_wmo_monthly_stats.sh,
export COMIN=/lfs/h2/emc/vpppg/noscrub/emc.vpppg/$NET/$evs_ver_2d
**add** export VDATE=20250930

Then, qsub  jevs_global_det_gfs_atmos_wmo_monthly_stats.sh
The job takes about 3 hours to finish

Commit:
add observation number n to rec2 file and change "parm" to "par" in svs
file.

QiShi-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/wmo

QiShi-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/wmo

QiShi-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/wmo

QiShi-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/wmo

AliciaBentley-NOAA commented:
@QiShi-NOAA I have started testing jevs_global_det_gfs_atmos_wmo_monthly_stats.sh for VDATE=20250930. I think it will take ~3 hours to run, as you said. You can find the output at the following paths. Please check the results and the log files when the job completes. 

### jevs_global_det_gfs_atmos_wmo_monthly_stats.sh
**Log:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr848test/EVS/dev/drivers/scripts/stats/global_det/jevs_global_det_gfs_atmos_wmo_monthly_stats_00.o221927657
**DATA:** /lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/jevs_global_det_gfs_atmos_wmo_monthly_stats_00.221927657.cbqs01/
**Stats:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/stats/global_det/

CC @AndrewBenjamin-NOAA @SamiraArdani-NOAA

QiShi-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/wmo

QiShi-NOAA commented:
@AliciaBentley-NOAA The WMO files are successfully generated: key n is add to *rec2 files and "parm" is changed to "par" in *svs files. No errors in log files.

QiShi-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/wmo

AliciaBentley-NOAA commented:
@QiShi-NOAA Thanks! I see that the job finished without any errors or warnings!
```
### Job 221927657.cbqs01 - Exit status is 0
    resources_used.walltime = 03:04:35
    Resource_List.walltime = 03:30:00
```

The last time we talked to ECMWF, they said that our WMO grid2grid verification was in the correct format and looked correct, but that our grid2obs verification didn't look correct. The line was too flat, if I recall correctly. Is that the last thing that we heard from ECMWF about this? 

If we have not corrected our WMO grid2obs verification since we last spoke to ECMWF, I think that we need to remove the WMO grid2obs verification from EVS v2.0. I don't us to produce incorrect WMO grid2obs stats in prod. Both WMO stats jobs may need to be updated to remove the grid2obs stats. Please let me know if the WMO grid2obs stats are still incorrect and should be removed. Thanks!

CC @AndrewBenjamin-NOAA @sbanihash

QiShi-NOAA commented:
@AliciaBentley-NOAA I confirm that the WMO stats for grid2obs are still incorrect. I'll work on removing them.

Commit:
turn off the surface stats for WMO

Commit:
Merge branch 'feature/wmo' of github.com:QiShi-NOAA/EVS into feature/wmo

QiShi-NOAA commented:
@AliciaBentley-NOAA I've turned off grid2obs_upperair and grid2obs_sfc in the following script:
global_det_atmos_stats_wmo_create_daily_job_scripts.py
global_det_atmos_stats_wmo_create_monthly_job_scripts.py

For daily WMO job, only wmo stats file is created, and no station_info file is created
For monthly job, only rec2 files are created, and no svs files are created.
Both jobs take much shorter time to finish.

AliciaBentley-NOAA commented:
@QiShi-NOAA Thank you! I have pulled in your updates and started testing the two WMO stats scripts (daily/monthly). We can have you reduce the walltimes after we see how long these two jobs take to run in these tests!

1) jevs_global_det_gfs_atmos_wmo_daily_stats.sh for VDATE=20251015
2) jevs_global_det_gfs_atmos_wmo_monthly_stats.sh for VDATE=20250930

**Logs:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr848test/EVS/dev/drivers/scripts/stats/global_det/
**DATA:** /lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/
**Stats:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/stats/global_det/

CC @AndrewBenjamin-NOAA @SamiraArdani-NOAA @sbanihash

Commit:
add check existence before remove grid2obs_upperair in the list

AliciaBentley-NOAA commented:
@QiShi-NOAA I see that you pushed a commit to your WMO PR after the last time that ran the two WMO jobs. Can you please explain what your new commit is and which WMO jobs I should rerun? Thanks!

New commit was called: `"add check existence before remove grid2obs_upperair in the list"`

CC @AndrewBenjamin-NOAA @sbanihash

QiShi-NOAA commented:
@AliciaBentley-NOAA I checked the log file from the original run, and find an error message related to remove grid2obs job from the jobs list without checking if the grib2obs job exist in the job list. The commit has address the issue. Please rerun the daily job:
global_det_atmos_stats_wmo_create_daily_job_scripts.py

QiShi-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/wmo

AliciaBentley-NOAA commented:
Thanks, @QiShi-NOAA! I have pulled in your changed and decided to rerun both jobs. Both jobs will be running for a long time. Please check back later (or tomorrow) to examine the log files and output. Thanks!

jevs_global_det_gfs_atmos_wmo_daily_stats.sh  (VDATE=20251020)
jevs_global_det_gfs_atmos_wmo_monthly_stats.sh  (VDATE=20250930)

**Logs:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr848test/EVS/dev/drivers/scripts/stats/global_det/
**DATA:** /lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/
**Stats:** /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/stats/global_det/

CC @AndrewBenjamin-NOAA @SamiraArdani-NOAA @sbanihash

AliciaBentley-NOAA commented:
@QiShi-NOAA Both jobs completed very early without any errors or warnings. Can you please examine the output and let me know if things look good? Also, if things do look good, can you please update the walltimes of these two jobs? 

```
jevs_global_det_gfs_atmos_wmo_daily_stats_00.o222638322:### Job 222638322.cbqs01 - Exit status is 0
jevs_global_det_gfs_atmos_wmo_daily_stats_00.o222638322:    resources_used.walltime = 00:01:04
jevs_global_det_gfs_atmos_wmo_daily_stats_00.o222638322:    Resource_List.walltime = 01:10:00
```
**Update walltime to 00:10:00, if the final .stat file contains the correct grid2grid values?**
Stats: /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/stats/global_det/gfs.20251020

```
jevs_global_det_gfs_atmos_wmo_monthly_stats_00.o222637708:### Job 222637708.cbqs01 - Exit status is 0
jevs_global_det_gfs_atmos_wmo_monthly_stats_00.o222637708:    resources_used.walltime = 00:29:24
jevs_global_det_gfs_atmos_wmo_monthly_stats_00.o222637708:    Resource_List.walltime = 03:30:00
```
**Update walltime to 00:40:00, if the final .rec2 files contains the correct grid2grid values?**
Stats: /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/stats/global_det/gfs.20250930

CC @AndrewBenjamin-NOAA @SamiraArdani-NOAA @sbanihash

QiShi-NOAA commented:
@AliciaBentley-NOAA the daily files all contains the correct grid2grid values comparing to emc.vpppg. However, the monthly file has extra grid2obs values with "nil" value.
evs/v2.0/stats/global_det/gfs.20250930/202509_kwbc_monthly.rec2
centre=kwbc,model=gfs,d=202509,ref=**ob,**par=w850hPa,sc=me,dom=asia,t=12,s=228,n=30,v=**nil**

I'm looking into the code now.

Commit:
turn off output grid2obs_upperair monthly stats

Commit:
Merge branch 'feature/wmo' of github.com:QiShi-NOAA/EVS into feature/wmo

Commit:
modify the walltime

QiShi-NOAA commented:
@AliciaBentley-NOAA I've added a fix to not output grib2obs stats in monthly rec2 file. I've also updated wall time for both daily and monthly script. The new monthly run I tested used 00:37:15, so I set the wall time to 00:50:00.
Would you mind testing both daily and monthly job?

QiShi-NOAA committed:
Merge branch 'NOAA-EMC:develop' into feature/wmo

QiShi-NOAA commented:
@AliciaBentley-NOAA Would you mind pausing the test. I the final stats looks good. But the small stats still have surface files in the atmos folder. I'm working on turn them off.

AliciaBentley-NOAA commented:
@QiShi-NOAA Sure! No problem! Please let me know when things are in a place where I should pull in the new changes are start again. Thanks!

Commit:
turn off wms_station_info and svs file

Commit:
Merge branch 'feature/wmo' of github.com:QiShi-NOAA/EVS into feature/wmo

QiShi-NOAA commented:
@AliciaBentley-NOAA I've add checks in theglobal_det_atmos_stats_wmo_create_monthly_job_scripts.py script to trun off wmo station file and svs file. The code is ready to test now.

AliciaBentley-NOAA commented:
Thanks, @QiShi-NOAA! I have pulled in your changed and am rerunning both jobs. Please check back later (or tomorrow) to examine the log files and output. Thanks!

**jevs_global_det_gfs_atmos_wmo_daily_stats.sh (VDATE=20251022)
jevs_global_det_gfs_atmos_wmo_monthly_stats.sh (VDATE=20250930)**

Logs: /lfs/h2/emc/vpppg/noscrub/alicia.bentley/pr848test/EVS/dev/drivers/scripts/stats/global_det/
DATA: /lfs/h2/emc/stmp/alicia.bentley/evs_test/prod/tmp/
Stats: /lfs/h2/emc/vpppg/noscrub/alicia.bentley/evs/v2.0/stats/global_det/

CC @AndrewBenjamin-NOAA @SamiraArdani-NOAA @sbanihash

QiShi-NOAA commented:
@AliciaBentley-NOAA The output  looks good (only contain grid2grid_upperair stats). No errors are found in the log files.

ShannonShields-NOAA commented:
Hi @QiShi-NOAA As a second reviewer, I was reviewing the files changed and I noticed there were some dev/drivers changed, but no ecf changes. Are there associated ecf scripts for WMO?

Commit:
update ecf walltime

QiShi-NOAA commented:
@ShannonShields-NOAA Thank you for point that out. I've updated the ecf walltime for wmo job.

AliciaBentley-NOAA commented:
@QiShi-NOAA I have reviewed the log files from the latest runs. I don't see any errors or warnings. The walltimes are consistent with NCO standards. 
```
jevs_global_det_gfs_atmos_wmo_daily_stats_00.o222739662:### Job 222739662.cbqs01 - Exit status is 0
jevs_global_det_gfs_atmos_wmo_daily_stats_00.o222739662:    resources_used.walltime = 00:01:03
jevs_global_det_gfs_atmos_wmo_daily_stats_00.o222739662:    Resource_List.walltime = 00:10:00

jevs_global_det_gfs_atmos_wmo_monthly_stats_00.o222739711:### Job 222739711.cbqs01 - Exit status is 0
jevs_global_det_gfs_atmos_wmo_monthly_stats_00.o222739711:    resources_used.walltime = 00:05:47
jevs_global_det_gfs_atmos_wmo_monthly_stats_00.o222739711:    Resource_List.walltime = 00:15:00
```
The output looks good to me. The daily WMO stat file is considerably smaller than in the emc.vpppg parallel, suggesting to me that only the grid2grid upper air stats are included. 
```
Test: -rw-r--r--  1 alicia.bentley vpppg 6.1M Oct 23 19:56 evs.stats.gfs.atmos.wmo.v20251022.stat
Para: -rw-r--r--    1 emc.vpppg vpppg 4.5G Oct 23 19:44 evs.stats.gfs.atmos.wmo.v20251022.stat
```
In the gfs.20250930 directory, the monthly file is much smaller than the daily file (as we would expect). 
```
-rw-r--r--  1 alicia.bentley vpppg  89M Oct 23 20:01 202509_kwbc_daily.rec2
-rw-r--r--  1 alicia.bentley vpppg 2.9M Oct 23 20:00 202509_kwbc_monthly.rec2
```
I'll go ahead and do my final code review! CC @ShannonShields-NOAA

AliciaBentley-NOAA reviewed (approved):
I have reviewed the coding changes made in this PR and I think that they look good. This PR was updated after submission to only include the 'grid2grid_upperair' verification in the WMO scripts. Note that only the .rec2 files are produced when only running 'grid2grid_upperair'. The .svs files (grid2obs) will come back later. The daily and monthly log files do not have errors or warnings. I approve this PR to be merged. 

@QiShi-NOAA Can you please confirm that your fork is synced?

CC @ShannonShields-NOAA @sbanihash

QiShi-NOAA commented:
@AliciaBentley-NOAA I confirm that this fork is synced.

ShannonShields-NOAA reviewed (approved):
@AliciaBentley-NOAA and @QiShi-NOAA I have reviewed the code changes to fix/remove g2o in WMO stat files and after several rounds of edits, the tests were successful. Walltimes were reduced to meet standards. I approve this PR to be merged.
